src/Layer/VSkiaContext.hpp

```diff
@@ -97,16 +100,41 @@ class SkiaContext
   void resizeSurface(int w, int h)
   {
     ASSERT(m_skiaSurfaceCreateProc);
+    if (m_surfaceExtent)
+    {
+      auto [width, height] = *m_surfaceExtent;
+      if (width == w && height == h)
+        return;
+    }
     m_surface = m_skiaSurfaceCreateProc(m_grContext.get(), w, h, m_ctxConfig);
+    m_surfaceExtent = { w, h };
+    if (!m_surface)
+      DEBUG("In resizeSurface: Recreate surface failed");
   }
 
   bool flushAndSubmit()
   {
     auto ok = m_grContext->submit();
     m_grContext->flush();
+    m_surface = nullptr; // invalid surface
     return ok;
   }
 
+  bool prepareFrame()
+  {
+    if (m_surfaceExtent && !m_surface)
+    {
+      auto [w, h] = *m_surfaceExtent;
+      VGG::layer::Timer t;
+      m_surface = m_skiaSurfaceCreateProc(m_grContext.get(), w, h, m_ctxConfig);
```

Comment:
We do not need to recreate surface each frame unless it is a wrapper of MTK view.

Reply:
According to https://groups.google.com/g/skia-discuss/c/3c10MvyaSug, the surface recreation cost is low.